### PR TITLE
Word count improvement fixes

### PR DESCRIPTION
**Pull request by cddr**
A few stylistic suggestions to #102 

  - make topic-metadata [the best function it can be](https://youtu.be/YR5WdGrpoug?t=1066)
  - update the test so that it closes the driver
  - add an "empty-state" fixture that is nice to have in-case there's any state left-over from a failed test-run

### CODE REVIEW

File: src/jackdaw/test/fixtures.clj [hunk 1]
```diff
@@ -80,6 +81,25 @@
...
+(defn empty-state-fixture [app-config]
+  (fn [t]
+    (let [state-dir (format "%s/%s"
+                            (or (get app-config "state.dir")
+                                "/tmp/kafka-streams")
+                            (get app-config "application.id"))]
+      (when (.exists (io/file state-dir))
+        (log/info "deleting state dir: " state-dir)
+        (delete-recursively state-dir))
+      (t))))
```
Comment (creese) [inline src/jackdaw/test/fixtures.clj]: Can we move these fixtures to the other PR?

File: examples/word-count/src/word_count.clj [hunk 2]
```diff
@@ -5,15 +5,14 @@
...
-            [jackdaw.streams :as j]))
-
+  (:require
+   [clojure.string :as str]
+   [clojure.java.io :as io]
+   [clojure.tools.logging :refer [info]]
+   [jackdaw.serdes.edn :as jse]
+   [jackdaw.serdes.resolver :as resolver]
+   [jackdaw.streams :as j])
+  (:gen-class))
```
[the remark this answers is not in the document]
Comment (cddr) [reply examples/word-count/src/word_count.clj]: https://stuartsierra.com/2016/clojure-how-to-ns.html

File: examples/word-count/src/word_count.clj [hunk 3]
```diff
@@ -42,20 +41,21 @@
   (resolver/serde-resolver))
 
 (def topic-metadata
-  "Returns the topic metadata map with serdes resolved."
-  (memoize (fn []
-             (fmap #(assoc % :key-serde (resolve-serde (:key-serde %))
-                           :value-serde (resolve-serde (:value-serde %)))
-                   +topic-metadata+))))
+  (reduce-kv (fn [m k v]
```
[the remark this answers is not in the document]
Comment (cddr) [reply examples/word-count/src/word_count.clj]: `fmap` requires a dependency that is only required for this one thing. Why carry the dependency when there is a core function that does the job?

[the remark this answers is not in the document]
Comment (creese) [reply examples/word-count/src/word_count.clj]: The code is more readable.

File: examples/word-count/src/word_count.clj (hunk 2 above)
[the remark this answers is not in the document]
Comment (creese) [reply examples/word-count/src/word_count.clj]: This seems largely reasonable. I was mostly wondering about the recommendation re line breaks which which to me seemed somewhat arbitrary. However, he gives a good reason:
> This recommendation is slightly different from common practice, but it makes it easier to sort names with editor commands (sort-lines in Emacs).

**Pull request merged**